technique/muon/zero_field_setup.py

```diff
@@ -0,0 +1,278 @@
+from genie_python import genie as g
+import time
+import numpy as np
+import matplotlib.pyplot as plt
+from scipy import stats
+from builtins import range
+from builtins import input
+
+PV_AUTO_FEEDBACK_MODE = "TE:NDLT1205:ZFCNTRL_01:AUTOFEEDBACK"
+PV_MAGNETIC_FIELD_STRENGTH = "TE:NDLT1205:ZFMAGFLD_01:FIELDSTRENGTH"
+
+PV_SETPOINT_X = "TE:NDLT1205:ZFCNTRL_01:FIELD:X:SP"
+PV_SETPOINT_Y = "TE:NDLT1205:ZFCNTRL_01:FIELD:Y:SP"
+PV_SETPOINT_Z = "TE:NDLT1205:ZFCNTRL_01:FIELD:Z:SP"
+
+POWER_SUPPLY_MAX = 100.00
+
+
+class ZeroFieldSetupProcedure():
+    """
+    Class for setting up zero field procedure. Contains all the required function which can be run sequentially to
+    to complete the set up
+    """
+    # pylint: disable=W0232
+
+    def read_values(self):
+        """
+        Read corrected values from the OPI
+        :return: corrected values for all axis
+        """
+        x = self.get_single_corrected_field_value("X")
+        y = self.get_single_corrected_field_value("Y")
+        z = self.get_single_corrected_field_value("Z")
+
+        return x, y, z
+
+    def get_magnitude(self):
+        """
+        Reads magnitude value from the OPI
+        :return: magnitude of magnetic field
+        """
+        return g.get_pv(PV_MAGNETIC_FIELD_STRENGTH)
+
+    def set_setpoints_to_zero(self, value_x, value_y, value_z):
+        """
+        Set all set points to 0
+        :param value_x: target value for x axis
+        :param value_y: target value for y axis
+        :param value_z: target value for z axis
+        :return: None
+        """
+        g.set_pv(PV_SETPOINT_X, value_x)
+        g.set_pv(PV_SETPOINT_Y, value_y)
+        g.set_pv(PV_SETPOINT_Z, value_z)
+
+    def get_correlated_current_and_fields(self, plot=False):
+        """
+        Gets correlated current and corrected fields, 2 second wait before taking the field
+        reading in all three axis
+        :param plot: data to be plotted or not
+        :return: correlated current and fields for each axis
+        """
+        iteration_number = 21
+
+        current_x = []
+        current_y = []
+        current_z = []
+
+        fields_x = []
+        fields_y = []
+        fields_z = []
+
+        # after every 5% increase in current, wait for 2 second and read the field
+        # in all three dimensions
+        for x in range(iteration_number):
+            current_x.append(((x - 10) / 10.0) * POWER_SUPPLY_MAX)
+            current_y.append(((x - 10) / 10.0) * POWER_SUPPLY_MAX)
+            current_z.append(((x - 10) / 10.0) * POWER_SUPPLY_MAX)
```

Comment:
Check the newly updated steps in the wiki. We shouldn't be hardcoding 10 as we actually just want 21 evenly-spaced steps.

Reply:
Fixed.